vng_api_common/middleware.py

```diff
@@ -96,6 +101,111 @@ def has_scopes(self, scopes: Union[Scope, None]) -> bool:
         return scopes.is_contained_in(scopes_provided)
 
 
+class JWTAuth:
+    def __init__(self, encoded: str = None):
+        self.encoded = encoded
+
+    @property
+    def applicaties(self) -> Union[list, None]:
+        if self.client_id is None:
+            return []
+
+        applicaties = self._get_auth()
+
+        if not applicaties:
+            auth_data = self._request_auth()
+            applicaties = self._save_auth(auth_data)
+
+        return applicaties
+
+    def _request_auth(self) -> list:
+        client = AuthorizationsConfig.get_client()
+
+        response = client.list(
+            'applicatie',
+            query_params={'client_ids': self.client_id}
+        )
+        return underscoreize(response['results'])
+
+    def _get_auth(self):
+        return Applicatie.objects.filter(client_ids__contains=[self.client_id])
+
+    def _save_auth(self, auth_data):
+        applicaties = []
+
+        for applicatie_data in auth_data:
+            applicatie_serializer = ApplicatieSerializer(data=applicatie_data)
+            applicatie_serializer.is_valid()
+            applicaties.append(applicatie_serializer.save())
+
+        return applicaties
+
+    @property
+    def client_id(self) -> Union[str, None]:
+        if self.encoded is None:
+            return None
+
+        # TODO extract client_id from payload
```

Comment:
:+1: 

We should probably support both for a while - check the payload first, if not found in payload, look in the header.

Logging that a client is using the old format should also be done so we can warn clients (at some point).

Reply:
added try-except block with extraction `client_id` from payload